packages/ra-ui-materialui/src/input/NumberInput.tsx

```diff
@@ -79,12 +79,15 @@ const NumberInput: FunctionComponent<NumberInputProps> = ({
                 />
             }
             label={
-                <FieldTitle
-                    label={label}
-                    source={source}
-                    resource={resource}
-                    isRequired={isRequired}
-                />
+                label !== '' &&
+                label !== false && (
+                    <FieldTitle
```

Comment:
It would probably be better to handle this in `FieldTitle` directly. Then we would only have to update the propTypes in all the other components

Reply:
Handled label issue directly inside `FieldTitle`. Updated propType for NumberInput field.
I will raise separate PR to update label propType in other components. Will it be fine?
Let me know if any  changes required.
Thanks..!

Reply:
Please update the propTypes and TypeScript types in this PR